### PR TITLE
Make the address 2 line visible when a value is autofilled

### LINKED ISSUE #47586: [Enhancement]: Make autofill compatible with updated address fields display logic
### Describe the solution you'd like

In #47160, we've updated the address fields display logic. As a result of that, it's now possible that the `Address line 2` field is hidden behind a link. In this case, autofill would not work for this field, as it only becomes available when the link is clicked. It's also not available in the DOM. This issue aims to enable autofill for the hidden `Address line 2` field. In https://github.com/woocommerce/woocommerce/pull/47160#pullrequestreview-2060269899, @senadir mentioned the following:

> One way we did autofill before was to render a hidden input (not sure if type hidden or just visually hidden) that would capture the autofill value, if captured, it will expand the actual field and fill it with that value.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by opr**
### Changes proposed in this Pull Request:

Adds a hidden (hidden by CSS, not actually hidden) field to the `AddressLine2Field` component to capture autofill events.
Before this change, the field did not exist in the DOM so autofill programmes could not enter text.

The reason it is hidden by CSS and not just a `type="hidden"` field is because while testing, the hidden field was ignored by autofill.

Closes #47586

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Have a saved address set up in your browser. Make sure this saved address contains an entry for address 2. (A tip to get this working is to fill in the checkout form with the data and check out on the site, then your browser should offer to save the address info for you).
2. In a new session (or incognito window) add an item to your cart and go to the Checkout block.
3. Ensure address 2 is not visible, but the link to add it is. Do not click it.
4. Use autofill to enter the address.
5. Ensure the address line 2 is autofilled correctly and is visible.
6. Check out and then check the order details to ensure address line 2 was recorded correctly.
7. Try removing address line 2 from your saved address and repeat these steps. The field should *not* become visible if address 2 is not in the autofill data.
8. Try in different browsers and on mobile.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Comment by ralucaStan** [comment]
@nielslange we need to have this ship with 9.2. Could you prioritize reviewing it?

**Comment by nielslange** [comment]
> @nielslange we need to have this ship with 9.2. Could you prioritize reviewing it?

I'm reviewing it now, @ralucaStan.

File: plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/address-line-2-field.tsx [hunk 1]
```diff
@@ -20,10 +21,18 @@ const AddressLine2Field = < T extends AddressFormValues | ContactFormValues >( {
 	const isFieldRequired = field?.required ?? false;
 
 	// Display the input field if it has a value or if it is required.
-	const [ isFieldVisible, setFieldVisible ] = useState(
+	const [ isFieldVisible, setIsFieldVisible ] = useState(
```
Comment (nielslange) [inline plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/address-line-2-field.tsx:24]: Good catch!

**Review by nielslange — APPROVED** [review]
Thanks for working on this, @opr. I've successfully tested the PR, with and without the `address line 2 field` value, against:
- Chrome on macOS
- Firefox on macOS
- Opera on macOS
- Safari on macOS
- Chrome on Android

**Review by senadir — APPROVED** [review]
Works as expected!

**Pull request merged**